docs/rfcs/UserProfilesv2_schema.json

```diff
@@ -0,0 +1,486 @@
+      "type": "string",
+      "title": "The Shirtsize Schema ",
+      "description": "The American shirt-size for the user.",
+      "default": "",
+      "examples": [
+        "Fitted S", "Straight-cut M", "Fitted L", "Fitted XL", "Straight-cut XXL", "Fitted XXXL"
+      ]
+    },
+    "groups": {
+      "$id": "/properties/groups",
+      "type": "array",
+      "items": {
+        "$id": "/properties/groups/items",
+        "type": "string",
+        "title": "The 0th Schema ",
+        "description": "User group membership utilized for access control. This structure is used for compatibility and SHOULD NOT be used. Even so, there is NO plan to deprecate this structure. Use `accessInformation` instead. Non-prefixed groups are known as Mozilla LDAP-groups. Groups prefixed by a provider name and an underscore are provided by a specific group publisher, such as `mozillianorg_` and `hris_`. See https://github.com/mozilla-iam/cis/blob/master/docs/Profiles.md for more information.",
+        "default": "0A_legacy_structure_only_please_use_accessInformation_instead",
+        "examples": [
+          "0A_legacy_structure_only_please_use_accessInformation_instead",
+          "mozilliansorg_bar",
+          "mozilliansorg_foo",
+          "hris_foo",
+          "ldapfoo"
+        ]
+      }
+    },
+    "accessInformation": {
+      "$id": "/properties/accessInformation",
+      "type": "object",
+      "properties": {
+        "ldap": {
+          "$id": "/properties/accessInformation/properties/ldap",
+          "type": "array",
+          "items": {
+            "$id": "/properties/accessInformation/properties/ldap/items",
+            "type": "object",
+            "properties": {
+              "name": {
+                "$id": "/properties/accessInformation/properties/ldap/items/properties/name",
+                "type": "string",
+                "title": "The Name Schema ",
+                "description": "Access information for Mozilla LDAP. This list all groups in Mozilla LDAP that this user is member of.",
+                "default": "",
+                "examples": [
+                  "ldapfoo"
+                ]
+              }
+            },
+            "required": [
+              "name"
+            ]
+          }
+        },
+        "mozilliansorg": {
+          "$id": "/properties/accessInformation/properties/mozilliansorg",
+          "type": "array",
+          "items": {
+            "$id": "/properties/accessInformation/properties/mozilliansorg/items",
+            "type": "object",
+            "properties": {
+              "name": {
+                "$id": "/properties/accessInformation/properties/mozilliansorg/items/properties/name",
+                "type": "string",
+                "title": "The Name Schema ",
+                "description": "Access information Mozillians.org (`mozilliansorg_` in `groups`). This list all access groups in Mozillians.org that this user is member of.",
+                "default": "",
```

Comment:
A "pattern" documenting the characters allows here would be nice!

I know we rely on LDAP and mozillians group names consistenting of certain character classes, ASCII printable I believe (without *).

It would be neat to see these character classes documented and validated.

Probably `"pattern": "^[a-zA-Z0-9_-]+$"` is fairly safe to add here... And it gives some super useful invariants for downstream consumers like me :)

Reply:
@nemo @tasos do you have any such filter on group names in mozillians.org right now? I agree it'd be nice to have this and we could standardize on something. LDAP is a-Z0-9 so far, to my knowledge (@jdow?)

Reply:
It doesn't have to be the same for both LDAP and mozillians... That would be great...

But just documenting what character classes are permitted... Probably also documenting the maximum and minimum string lengths...

It's nice to know that my services will never have to deal with the mozillians group named empty string... Or an LDAP group with a name that is 9000 chars long... Those probably won't happen, so let's document the limits as it reduces the corner cases downstream consumers have to deal with.

In reality downstream consumers won't deal with extreme corner cases I outline above. But documenting what they get will it more likely they handle all inputs correctly.

Reply:
do you know how to do string len? @jonasfj